src/components/layerpublisher/LayerPublisher.js

```diff
@@ -43,6 +45,36 @@ function info(fileName) {
   PubSub.publish(STATUS_INFO, `Loaded ${fileName}.`);
 }
 
+async function initRasterLayer(data) {
+  const channelNames = Object.keys(data);
+  const channelUrls = Object.values(data).map(d => d.tileSource);
+  const raster = {
+    channelNames,
+    domains: Object.values(data).map(d => d.range),
+    id: String(Date.now()),
+  };
+  if (channelUrls[0].includes('tif')) {
+    const loader = await createTiffPyramid({ channelNames, channelUrls });
+    return { ...raster, loader };
+  } if (channelUrls[0].includes('zarr')) {
+    const rootZarrUrl = channelUrls[0].slice(0, -1);
```

Comment:
```suggestion
    const rootZarrUrl = channelUrls[0].slice();
```
suffices to make a copy...

Reply:
Not quite, there is an issue with the URL currently that requires truncating it to remove the last backslash. This was hidden within viv before, and will be fixed with `vitessce-image-loader`

Reply:
Again moving all this hardcoded logic within `initRasterLayer` makes identifying these issues very clear IMO.

Reply:
Cool -- Sounds good, as long as it's on your radar.

(In general, I'm ok with TODOs in the code, as long it's not a substitute for the actual issue. The ideal might be to have an issue, and a TODO with a comment and a URL for the issue... but that's often going to be overkill.)